src/org/zaproxy/zap/extension/script/ProxyScript.java

```diff
@@ -24,9 +24,44 @@
 
 import org.parosproxy.paros.network.HttpMessage;
 
+/**
+ * A script that will be executed when a proxied request is ready to be forwarded to the server and when the response is ready
+ * to be forwarded to the client.
+ * <p>
+ * <strong>Note:</strong> In the presence of more than one script or listener (for example,
+ * {@link org.parosproxy.paros.core.proxy.ProxyListener ProxyListener} and {@link org.zaproxy.zap.network.HttpSenderListener
+ * HttpSenderListener}) there are <i>no</i> guarantees that the (final) request/response is exactly the same as the one crafted
+ * by this script, or, that the request/response is actually going to be forwarded, as the following scripts/listeners may
+ * modify it or drop it.
+ *
+ * @since 2.2.0
+ */
 public interface ProxyScript {
 
+	/**
+	 * Called when a request is received from the client.
+	 * <p>
+	 * Both the request and response of the HTTP message may be modified. If a response is set the request is <i>not</i>
+	 * forwarded to the server, thus preventing the request and overriding the response. If the return value is {@code true} the
+	 * request <i>may be</i> forwarded and the following/listeners will be called, if the value is {@code false} the request is
```

Comment:
> following/listeners

Is the slash needed or is a word missing?

Reply:
Word missing, "scripts".

Reply:
Added.